### PR TITLE
Number of updates to "Identity for Developers" hack based on recent delivery

### PR DESCRIPTION
**Pull request by jelledruyts**
See individual commits for details :-)

### CODE REVIEW

**Comment by jelledruyts** [comment]
@DavidHoerster as you've reviewed [my colleague's earlier PR](https://github.com/microsoft/WhatTheHack/pull/251) - would you like to take a look at my proposed changes here as well? :-)

**Review by DavidHoerster — APPROVED** [review]
Looks good - like the updates / improvements!

**Comment by jelledruyts** [comment]
Sorry, accidentally requested another review from David but he had already approved :-)

File: 021-IdentityForApps/Student/04-l14n.md [hunk 1]
```diff
@@ -10,15 +10,16 @@ CMC IT Leadership is _extremely_ impressed with your SignUp / SignIn flow and ho
 
 First off, one of your developers has developed a _harness application_ where you can incorporate your SignUp / SignIn User Flow. Your developer has parameterized it so that you should only have to modify your application's settings in order to incorporate the User Flow.
 
-**NOTE:** This web application is located in the folder named **HarnessApp** in your Teams channel Files tab. 
+**NOTE:** This web application is located in the **HarnessApp** folder under the [student resources](Resources/HarnessApp) or in your Teams channel Files tab.
```
Comment (jrzyshr) [inline 021-IdentityForApps/Student/04-l14n.md:13]: The student guide should never link directly to the WTH repo!  This is to keep the students from opening the repo and easily accessing the /Coach folder and the solution guide. :)  

As per the WTH Author's guide, you should refer to the files being in the Resources.zip file provided by their coach (typically in the File tab of your Teams channel).  You may refer to the relative path within the Resources.zip file which is the same as the relative path within /Student/Resources folder.

This hack does refer to the students to find resources in the Files tab of their Teams channel.  That is fine.  Please remove the direct link to the repo.  

If you want to make the language clearer, consider the following text:

"This web application is located in the **HarnessApp** folder within the Resources.zip file provided by your coach."
or
"This web application is located in the **HarnessApp** folder within the Resources.zip file provided by your coach or in the Files tab of your Teams channel."

File: 021-IdentityForApps/Student/04-l14n.md [hunk 2]
```diff
@@ -10,15 +10,16 @@ CMC IT Leadership is _extremely_ impressed with your SignUp / SignIn flow and ho
...
 
 ### Custom Templates and Themes
 
 CMC IT Leadership has decided that, while Slate Gray is a nice theme and all, it's pretty plain. They have asked you to add some custom branding, such as corporate logo and background images, and also to use a different layout for the SignUp / SignIn flow.
 
 CMC IT Leadership would like you to use the provided _HTML layout templates, styles, and images_. You will not need to modify the template unless you want, but you will have to find a place to host these resources. Also, examine the resources in this folder and understand how the templates, CSS files, and images are related. There's a marker for a storage account name in several files, so you may need to replace that at some point.
 
-**NOTE:** These files are located in the folder named **PageTemplates** in your Teams channel Files tab.
+**NOTE:** These files are located in the **PageTemplates** folder under the [student resources](Resources/PageTemplates) or in your Teams channel Files tab.
```
Comment (jrzyshr) [inline 021-IdentityForApps/Student/04-l14n.md:21]: As per the comment above, please remove the link to the WTH repo.

File: 021-IdentityForApps/Student/05-claims-enrichment.md [hunk 3]
```diff
@@ -26,7 +26,8 @@ You will call the Verify-inator at its `/Territory` endpoint. So if you deployed
 
 Lastly, CMC IT Leadership does not want to present the "Territory Name" attribute to the user during sign-up. They have asked you to please remove this field from the sign-up experience.
 
-**NOTE:** This ASPNETCORE web api application is located in the folder named **Verify-inator** in your Teams channel Files tab.
+**NOTE:** This ASPNETCORE web api application is located in the **Verify-inator** folder under the [student resources](Resources/Verify-inator) or in your Teams channel Files tab.
```
Comment (jrzyshr) [inline 021-IdentityForApps/Student/05-claims-enrichment.md:29]: As per the WTH Author's guide, please remove the link to the folder in the WTH repo.   Instead, refer to the relative path in the Resources.zip file provided by the coach.

File: 021-IdentityForApps/Student/05-claims-enrichment.md [hunk 4]
```diff
@@ -64,7 +65,7 @@ Your UserFlow will update all claims that are returned (such as TerritoryName) a
 CMC IT Leadership considers success in dealing with this QA issue if you are able to:
 
 - Create the new custom attribute for Territory Name in your B2C tenant (named "TerritoryName");
-- Deploy the Azure function (the "CMC ID Verify-inator"), code located in [Resources](./Resources/Verify-inator) ;
+- Deploy the "CMC ID Verify-inator" code located in [Resources](./Resources/Verify-inator);
```
Comment (jrzyshr) [inline 021-IdentityForApps/Student/05-claims-enrichment.md:68]: Please remove direct link to the WTH repo.  Instead, refer to the relative path in the Resources.zip file provided by the coach.

File: 021-IdentityForApps/Student/07-admin-graph.md [hunk 5]
```diff
@@ -12,7 +12,7 @@ Leadership would like you to build an app that can query the B2C tenant and outp
 
 Luckily, your innovative developer is still here (they haven't taken any time off either) and they've built a **console app** that can do some of this (but you'll need to make some modifications). Luckily, the developer has parameterized the B2C bits, so you'll have to just make some updates to the configuration settings file (appsettings.json) in order to connect to your B2C tenant.
 
-**NOTE:** This DOTNETCORE console application is located in the folder named **MSGraphApp** in your Teams channel Files tab.
+**NOTE:** This DOTNETCORE console application is located in the **MSGraphApp** folder under the [student resources](Resources/MSGraphApp) or in your Teams channel Files tab.
```
Comment (jrzyshr) [inline 021-IdentityForApps/Student/07-admin-graph.md:15]: Please remove direct link to the WTH repo. Instead, refer to the relative path in the Resources.zip file provided by the coach.

File: 021-IdentityForApps/Student/00-pre-reqs.md [hunk 6]
```diff
@@ -10,18 +10,19 @@ A smart cloud solution architect always has the right tools in their toolbox.
...
-    - `/Verify-inator` contains an ASPNETCORE Web Api application that will be used to validate user attributes from your Sign Up User Flow. You will need this in Challenge 5.
+- If applicable, make sure that you have joined the Teams group for this track. The first person on your team at your table should create a new channel in this Team with your team name.
+- Install the recommended toolset:
+  - [Azure Subscription](#azure-subscription)
+  - [Visual Studio Code](#visual-studio-code)
+    - [Visual Studio Code plugins for B2C](#visual-studio-code-plugins-for-arm-templates)
+  - [.NET Core 3.1 SDK](https://dotnet.microsoft.com/download/dotnet/3.1)
+  - [Managing Cloud Resources](#managing-cloud-resources)
+    - [Azure Portal](#azure-portal)
+- Locate and download the resources found in the "Files" tab of the Team's General channel. There are four folders each containing a different resource that you will use in various challenges:
```
Comment (jrzyshr) [inline 021-IdentityForApps/Student/00-pre-reqs.md:21]: In multiple locations, direct links were provided to these folders in the WTH repo.  These links should be removed as per the inline comments in this review.

Consider rewording this line to say:
"Locate and download the Resources.zip file found in the "Files" tab of the Team's General channel.  There are four folders in the zip file, each containing different resources that you will use in various challenges.

**Review by jrzyshr — CHANGES_REQUESTED** [review]

**Comment by jrzyshr** [comment]
@jelledruyts I requested some changes to remove links to the WTH repo.  I will trust @DavidHoerster's review for all of the technical details & code/config files.

**Comment by jelledruyts** [comment]
@jrzyshr before I roll back those changes I just wanted to double check where the harm is, as these are all relative links - so no direct link to the WTH repo, and if you download the repo locally (as a coach perhaps, and share the Student folder only with the group) they will just resolve to the local relative files.

On top of that, in my delivery, I _did_ in fact point the attendees directly at the WTH content with the explicit note that they shouldn't "peek" at the Coach folder. But even if they did though, they're just cheating themselves out of a good learning opportunity and then that's their decision really: we're dealing with adult professionals at the end of the day, so I'd count on their own best judgment. So by sharing the repo directly it just made it much easier for me to deliver without a Teams channel or sharing zip files, and has the additional advantage (IMHO) that attendees can still follow and even contribute to the WTH repo afterwards as they know where it lives.

Happy to roll back if you, just wanted to clarify why I think this can make it easier for everyone involved :-)

**Comment by jrzyshr** [comment]
@jelledruyts students should be accessing WTH through our main website at: https://aka.ms/wth 

That website is rendered from the MD files using Github pages.  If you link to a relative path in the repo, it will open the repo in the browser UNLESS there is a "README.md" in the folder that your relative path points to.   Upon a second look, I see that there are README.md files in both the HarnessApp and MSGraphApp folders that you have linked.  Those DO render as webpages, so it's not as bad as I originally thought.  However, the PageTemplates & Verify-inator folders do not have readme files, so they open the repo.  

That 2nd look led me to another set of issues.  I'm guessing the sample app code in "/Student/Resources" was borrowed from another repo ([Azure-Samples](https://github.com/Azure-Samples/).  In the README files for both HarnessApp and MSGraphApp, they instruct users to "clone the repo"... but provide a link to where I assume the sample app lives in the Azure-Samples repo on Github.  We must have missed this in the original review.

Adding @DavidHoerster 

My final recommendation here is to remove the PageTemplate & Verify-inator links and replace with text that refers to the "Resources.zip" file provided by your coach.   The HarnessApp & MSGraphApp links should stay.  However, the readme files in BOTH of those folders should be updated to remove the clone instructions and replaced with instructions that refer to accessing the code in the Resources.zip file provided by the coach.   That way, the students are instructed to use the code in WTH repo, not the external repo (Azure-Samples) which we do not control.

**Comment by DavidHoerster** [comment]
Hey all - I'm sorry I missed this activity. I will take a look at the 2nd set of changes that @jrzyshr suggested tonight, unless someone is already working on this.

**Comment by jrzyshr** [comment]
@DavidHoerster If you update the Readme files for the HarnessApp and MSGraphApp to refer to the "Resources.zip file provided by your coach", you should be able to do those via a separate PR.

@jelledruyts for your changes, please make the ones recommended in my last comment. (remove PageTemplate & Verify-inator links and replace with text that refers to the "Resources.zip" file provided by your coach, add language about "resources.zip" to the Harness & MSGraph links... which are okay to stay pending David's changes.)   

Thanks!

File: 021-IdentityForApps/Student/04-l14n.md (hunk 1 above)
Comment (jelledruyts) [reply 021-IdentityForApps/Student/04-l14n.md]: Fixed!

File: 021-IdentityForApps/Student/04-l14n.md (hunk 2 above)
Comment (jelledruyts) [reply 021-IdentityForApps/Student/04-l14n.md]: Fixed!

File: 021-IdentityForApps/Student/05-claims-enrichment.md (hunk 3 above)
Comment (jelledruyts) [reply 021-IdentityForApps/Student/05-claims-enrichment.md]: Fixed!

File: 021-IdentityForApps/Student/05-claims-enrichment.md (hunk 4 above)
Comment (jelledruyts) [reply 021-IdentityForApps/Student/05-claims-enrichment.md]: Fixed!

File: 021-IdentityForApps/Student/07-admin-graph.md (hunk 5 above)
Comment (jelledruyts) [reply 021-IdentityForApps/Student/07-admin-graph.md]: Fixed!

File: 021-IdentityForApps/Student/00-pre-reqs.md (hunk 6 above)
Comment (jelledruyts) [reply 021-IdentityForApps/Student/00-pre-reqs.md]: Done!

**Review by jrzyshr — APPROVED** [review]
Thanks Jelle for these awesome updates!

**Pull request merged**